python/mspasspy/db/database.py

```diff
@@ -1518,229 +1724,188 @@ def _check_required(self, collection='site', keys=['lat','lon','elev','starttime
+                                newvalue=copied_metadata[k]
+                                message='Had to convert type of data with key='+k+' from '+str(type(oldvalue))+' to '+str(type(newvalue))
+                                mspass_object.elog.log_error(alg_name,message,ErrorSeverity.Complaint)
                             except Exception as err:
-                                # update is not aborted, but mark the mspass object as dead
-                                if update_metadata_def.is_required(k):
-                                    mspass_object.elog.log_error('update_metadata',
-                                        "cautious mode: Required attribute {} has type {}, forbidden by definition and unable to convert".format(k, type(copied_metadata[k])),
-                                        ErrorSeverity.Invalid)
-                                    has_fatal_error = True
-                                    mspass_object.kill()
-
-                        else:
-                            # otherwise, we could update this attribute in the metadata
-                            insert_dict[k] = copied_metadata[k]
-
-                    # pedantic mode: any type mismatch could end up killing the mspass object
-                    elif mode == "pedantic":
-                        if not isinstance(copied_metadata[k], update_metadata_def.type(k)):
-                            mspass_object.elog.log_error('update_metadata',
-                                "pedantic mode: attribute {} has type {}, forbidden by definition".format(k, type(copied_metadata[k])),
-                                ErrorSeverity.Invalid)
-                            has_fatal_error = True
-                            mspass_object.kill()
+                                message = 'Cannot update data with key=' + k +'\n'
+                                if save_schema.is_required(k):
+                                    message += ' Required key value could not be converted to required type='+str(save_schema.type(k))+' actual type='+str(type(copied_metadata[k]))
+                                else:
+                                    message += ' Value stored has type='+str(type(copied_metadata[k])+' which cannot be converted to required type='+str(save_schema.type(k)))+'\n'
+                                message += 'Data for this key will not be changed or set in the database'
+                                message += '\nPython error exception message caught:\n'
+                                message += str(err)
+                                mspass_object.elog.log_error(alg_name,
+                                           message,ErrorSeverity.Complaint)
+                                # The None arg2 cause pop to not throw
+                                # an exception if k isn't defined - a bit ambigous in the try block
+                                insertion_dict.pop(k,None)
```

Comment:
We probably don't need to include this line here though it doesn't hurt because when a type conversion is unsuccessful, the key-value pair would not be added into insertion_dict.

Reply:
My thinking was it was better to be safe. As the comment says any exception in the try block would land there so I thought it better to cover the possibility it had been set somehow before.  That may be being overly paranoid.

Reply:
Yeah, sure it would be safe if we do that. I am fine with that.